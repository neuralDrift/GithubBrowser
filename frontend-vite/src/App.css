body {
  background-color: #f5f7fa;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  color: #212529;
  margin: 0;
  padding: 0;
}

.App {
  text-align: center;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

header {
  background-color: #24292e; /* GitHub dark tone */
  color: #ffffff;
  padding: 1rem 0;
  font-size: 1.5rem;
  font-weight: bold;
}

form {
  max-width: 600px;
  margin: 2rem auto;
}

input[type="text"] {
  border-radius: 8px;
  border: 1px solid #ccc;
  padding: 0.6rem 1rem;
  width: 75%;
  font-size: 1rem;
}

button {
  border-radius: 8px;
  border: none;
  padding: 0.6rem 1rem;
  margin-left: 0.5rem;
  background-color: #007bff;
  color: white;
  font-weight: bold;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

button:hover {
  background-color: #0056b3;
}

.user-card {
  background: white;
  border-radius: 12px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  padding: 2rem;
  margin: 1rem auto;
  max-width: 600px;
  text-align: left;
}

.user-card img {
  border-radius: 50%;
  width: 120px;
  height: 120px;
  margin-bottom: 1rem;
}

.user-card h2 {
  font-size: 1.5rem;
  margin-bottom: 0.5rem;
}

.list-group {
  margin-top: 1rem;
}

.list-group-item {
  background: #fff;
  border: 1px solid #dee2e6;
  border-radius: 8px;
  margin: 0.5rem 0;
  padding: 1rem;
  transition: transform 0.2s ease;
}

.list-group-item:hover {
  transform: translateY(-2px);
}

a {
  text-decoration: none;
  color: #007bff;
}

a:hover {
  text-decoration: underline;
}

a.external-link {
  color: #28a745; /* Green for external links */
  font-weight: 500;
}

.spinner-border {
  margin-top: 3rem;
}

footer {
  margin-top: auto;
  padding: 1rem 0;
  background: #24292e;
  color: #ddd;
  font-size: 0.9rem;
}
