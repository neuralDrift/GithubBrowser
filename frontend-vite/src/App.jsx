// Main react entry point for frontend
// Sets up routing between pages/components via react-router-dom
import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";

// Import components
import UserSearch from "./components/UserSearch";
import UserDetails from "./components/UserDetails";
import RepoDetails from "./components/RepoDetails";

function App() {
  return (
    // Router wraps app to enable client-side routing
    <Router>
      <div className="container mt-4">
        <Routes>
          <Route path="/" element={<UserSearch />} />
          <Route path="/user/:username" element={<UserDetails />} />
          <Route path="/repo/:username/:repo" element={<RepoDetails />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
