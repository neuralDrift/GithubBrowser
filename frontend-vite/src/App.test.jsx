import React from "react";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import App from "./App.jsx";
import { BrowserRouter } from "react-router-dom";

describe("App component", () => {
  // Snapshot test
  it("matches snapshot", () => {
    // Render the App component
    const { asFragment } = render(<App />);
    // Capture the rendered output as a fragment and compare it to the saved snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
