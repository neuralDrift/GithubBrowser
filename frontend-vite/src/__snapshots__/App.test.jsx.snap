// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-4"
  >
    <div>
      <form>
        <div
          class="input-group mb-3"
        >
          <input
            class="form-control"
            placeholder="Search GitHub users (e.g., 'octo')"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
